File: src/main/kotlin/com/github/magneticflux/aoc/year2017/day1/Problem2.kt
package com.github.magneticflux.aoc.year2017.day1

import com.github.magneticflux.aoc.input
import com.github.magneticflux.aoc.lines

object Problem2 {
    @JvmStatic
    fun main(args: Array<String>) {
        val lineChars = input().lines().next().toCharArray()

        var count = 0

        lineChars.forEachIndexed { index, c ->
            if (lineChars[(index + lineChars.size / 2) % lineChars.size] == c)
                count += c.toString().toInt()
        }

        println("Count: $count")
    }
}

File: src/main/scala/com/github/magneticflux/aoc/year2018/day1/Problem2.scala
package com.github.magneticflux.aoc.year2018.day1

import com.github.magneticflux.aoc.Helpers._

import scala.collection.mutable
import scala.util.control.Breaks._

/**
  * Created by Mitchell Skaggs on 12/2/2018.
  */
object Problem2 {
  def main(args: Array[String]): Unit = {
    val numbers = this.input()
      .getLines()
      .map(_.toInt)
      .toSeq

    val infiniteNumbers = Iterator.continually(numbers).flatten

    val seenSoFar = mutable.Set[Int]()

    var total = 0

    breakable {
      for (i <- infiniteNumbers) {
        if (!seenSoFar.add(total)) {
          println(s"Found $total twice!")
          break
        }
        total += i
      }
    }
  }
}

